CMakeLists.txt

```diff
@@ -2020,6 +1988,44 @@ list(APPEND
   ${extra_post_build_byproducts}
   )
 
+if(CONFIG_BUILD_OUTPUT_BIN AND CONFIG_BUILD_OUTPUT_UF2)
+  if(CONFIG_BUILD_OUTPUT_UF2_USE_FLASH_BASE)
+    set(flash_addr "${CONFIG_FLASH_BASE_ADDRESS}")
+  else()
+    set(flash_addr "${CONFIG_FLASH_LOAD_OFFSET}")
+  endif()
+
+  if(CONFIG_BUILD_OUTPUT_UF2_USE_FLASH_OFFSET)
+    # Note, the `+ 0` in formula below avoids errors in cases where a Kconfig
+    #       variable is undefined and thus expands to nothing.
+    math(EXPR flash_addr
+        "${flash_addr} + ${CONFIG_FLASH_LOAD_OFFSET} + 0"
+         OUTPUT_FORMAT HEXADECIMAL
+    )
+  endif()
+
+  if(CONFIG_BOOTLOADER_MCUBOOT)
+    set(kernel_to_uf2_artifact "${KERNEL_NAME}.signed.bin")
+  else()
+    set(kernel_to_uf2_artifact "${KERNEL_BIN_NAME}")
+  endif()
+
+  list(APPEND
+    post_build_commands
+    COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/build/uf2conv.py
+            -c
+            -f ${CONFIG_BUILD_OUTPUT_UF2_FAMILY_ID}
+            -b ${flash_addr}
+            -o ${KERNEL_UF2_NAME}
```

Comment:
for signed bin files, the current convention is that they are named `<KERNEL_NAME>.signed.bin` and the unsigned image exists as `<KERNEL_NAME>.bin.
When signing then `zephyr.bin` + `zephyr.signed.bin` are both available, as one is produced from the other.

Similar for hex files where both `<KERNEL_NAME>.hex` and `<KERNEL_NAME>.signed.hex` are produced.

Converting the signed binary to uf2 is acceptable, but please make it clear that it is now a signed bin.
Meaning that the file name should be ``<kernel_name>.signed.uf2`, (default: `zephyr.signed.uf2`), so that it follows same naming pattern as the bin / hex files.

To avoid duplicating  logic which handles signing, then perhaps move the conversion to uf2 in here:
https://github.com/zephyrproject-rtos/zephyr/blob/4bb945300bf92e4708404102d2a3b54e540bfd6c/cmake/mcuboot.cmake#L194-L195

That will also help you if you need to handle confirmed and encrypted images in addition, as well as avoiding to clutter Zephyr's top CMakeLists.txt with signing related code.

Reply:
Thank you for your review!

Moving the UF2 conversion to the mcuboot cmake file sounds good. If our goal is to convert all of the possibly generated .bin files (signed, confirmed, ...) then maybe it's better if I move the UF2 conversion into a separate function and call that function whenever the build system generates a .bin file.

Reply:
do you mean a separate function inside `mcuboot.cmake` or a separate function in a separate file ?

I think a separate function inside `mcuboot.cmake` can make sense, question more becomes if we should then do the same for bin and hex output, so that the functions are aligned.

Having a separate function only for uf2 makes the implementation of the signing feature somewhat inconsistent.